PythonPonies/bangazon-api
# Zgl products on orders

# Description
Add description about what this pull request covers.
-This pull request covers expanding the Orders api endpoint to return information about products, products' catgories, and products' sellers.
## Number of Fixes
Type of number of fixes included in this PR.
- a small few on models and serializers
## Related Ticket(s)
List related ticket(s) that apply to the work done in this pull request
#78 #69 
## Problem to Solve
Describe the problem(s) this pull request solves.
- This fix solves issue of returning products information on the orders views. 
## Proposed Changes
Describe the proposed changes.
- Changed the Order model to have a products column that is a many to many field that traverses product on order
- Changed the Order and Product On Order Serializer to have different serializer depths
## Expected Behavior
Describe the expected behavior.
When you go to Orders, if an order has products, it should display the products, seller info, and product categories
## Steps to Test Solution

1. Git fetch
2. makemigrations
3. migrate
4. runserver
5. go to admin and add products to orders
6. go to api orders endpoint and see if products info

## Testing

[ ] There are new unit tests in this PR, and I verify that there is full coverage of all new code.
[ ] I certify that all existing tests pass

## Documentation
[ ] There is new documentation in this pull request that must be reviewed..
[ ] I added documentation for any new classes/methods

## Deploy Notes
Notes regarding deployment the contained body of work.  These should note any
db migrations, etc.
